src/wallet/wallet.cpp

```diff
@@ -2382,18 +2403,32 @@ CBlockIndex* CWallet::ScanForWalletTransactions(CBlockIndex *pindexStart, bool f
     CBlockIndex* pindex = pindexStart;
     {
         LOCK2(cs_main, cs_wallet);
+        // No need to read and scan block if block was created before our wallet birthday (as adjusted for block time variability).
+        // If you are recovering wallet with mnemonics, start rescan from the block when mnemonics were implemented in Firo.
+        // If the user provides a date, start scanning from the block that corresponds to that date.
+        // If no date is provided, start scanning from the mnemonic start block.
+
+        std::string wcdate = GetArg("-wcdate", "");
+        CBlockIndex* mnemonicStartBlock = chainActive[chainParams.GetConsensus().nMnemonicBlock];
```

Comment:
better to make sure the the result is not NULL like it was in old code 
something like this 
```
      pindex = chainActive[chainParams.GetConsensus().nMnemonicBlock];
      if (pindex == NULL)
          pindex = chainActive.Tip();
```